tests/unit/iontests.js

```diff
@@ -142,88 +82,280 @@ define(['intern', 'intern!object', 'intern/dojo/node!fs', 'intern/dojo/node!path
-            }
-        }
-
-        function makeEventStreamTest(path) {
-            return function() {
-                let executor = function(resolve, reject) {
-                    console.log(path);
-                    roundTripEventStreams(ion.makeReader(getInput(path)));
-                    resolve();
-                };
-
-                return new Promise(executor);
-            }
-        }
-
-        function getStreams(reader) {
+        function roundTripEventStreams(reader) {
             let streams = [];
             streams.push(new ion.IonEventStream(reader));
+
             let eventWriter = ion.makeTextWriter();
             streams[0].writeEventStream(eventWriter);
             eventWriter.close();
             streams.push(new ion.IonEventStream(new ion.makeReader(eventWriter.getBytes())));
+
             let textWriter = ion.makeTextWriter();
             streams[0].writeIon(textWriter);
             streams.push(new ion.IonEventStream(new ion.makeReader(textWriter.getBytes())));
+
             let binaryWriter = ion.makeBinaryWriter();
             streams[0].writeIon(binaryWriter);
             streams.push(new ion.IonEventStream(new ion.makeReader(binaryWriter.getBytes())));
+
             let eventTextWriter = ion.makeTextWriter();
             streams[0].writeIon(eventTextWriter);
             streams.push(new ion.IonEventStream(new ion.makeReader(eventTextWriter.getBytes())));
+
             let eventBinaryWriter = ion.makeBinaryWriter();
             streams[0].writeIon(eventBinaryWriter);
             streams.push(new ion.IonEventStream(new ion.makeReader(eventBinaryWriter.getBytes())));
-            return streams;
-        }
 
-        function roundTripEventStreams(reader) {
-            let streams = getStreams(reader);
-            //streams.push()
             for(let i = 0;  i < streams.length - 1; i++){
                 for(let j = i; j < streams.length; j++){
-                    if(!streams[i].equals(streams[(j + i) % streams.length])) throw new Error("Streams unequal.");
+                    if(!streams[i].equals(streams[(j + i) % streams.length])) {
+                        throw new Error("Streams unequal.");
+                    }
                 }
             }
         }
-
-        for (let file of goodUnskipped) {
-                    goodSuite[file] = makeGoodTest(file);
-                    eventStreamSuite[file] = makeEventStreamTest(file);
-        }
-        for (let file of badUnskipped) {
-                badSuite[file] = makeBadTest(file);
-        }
-
-        registerSuite(goodSuite);
-        registerSuite(badSuite);
-        registerSuite(eventStreamSuite);
     }
 );
 
+function toSkipList(paths) {
+    let skipList = {};
+    paths.forEach(path => {
+        skipList[path] = 1;
+    });
+
+    // additional, known bad/slow test files:
+    skipList['ion-tests/iontestdata/good/subfieldVarUInt32bit.ion'] = 1;
+
+    return skipList;
+}
+
+// skiplists generated by toSkipList.pl
```

Comment:
Is this comment referring to the function `toSkipList`? The `.pl` is what is confusing me here.

Reply:
it's a perl script i have locally;  removed reference to avoid confusion